lib/superagent-mock.js

```diff
@@ -18,12 +18,13 @@ function mock (superagent, config) {
    * Keep the default methods
    */
   var oldGet = superagent.get;
+  var oldPost = superagent.post;
   var oldEnd = Request.prototype.end;
 
   /**
-   * Override get function
+   * Attempt to match url against the patterns in the fixtures.
    */
-  superagent.get = function (url, data, fn) {
+  function testUrlForPatterns(url) {
```

Comment:
This function could return the parser value for the url and check before if `parsers[url]` isn't already filled, WYT ?